src/test/java/com/puppycrawl/tools/checkstyle/checks/OrderedPropertiesCheckTest.java

```diff
@@ -0,0 +1,228 @@
+////////////////////////////////////////////////////////////////////////////////
+// checkstyle: Checks Java source code for adherence to a set of rules.
+// Copyright (C) 2001-2019 the original author or authors.
+//
+// This library is free software; you can redistribute it and/or
+// modify it under the terms of the GNU Lesser General Public
+// License as published by the Free Software Foundation; either
+// version 2.1 of the License, or (at your option) any later version.
+//
+// This library is distributed in the hope that it will be useful,
+// but WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+// Lesser General Public License for more details.
+//
+// You should have received a copy of the GNU Lesser General Public
+// License along with this library; if not, write to the Free Software
+// Foundation, Inc., 59 Temple Place, Suite 330, Boston, MA  02111-1307  USA
+////////////////////////////////////////////////////////////////////////////////
+
+package com.puppycrawl.tools.checkstyle.checks;
+
+import static com.puppycrawl.tools.checkstyle.checks.OrderedPropertiesCheck.MSG_KEY;
+import static org.junit.Assert.assertEquals;
+import static org.junit.Assert.assertTrue;
+import static org.junit.Assert.fail;
+
+import java.io.File;
+import java.lang.reflect.Constructor;
+import java.lang.reflect.Method;
+import java.nio.charset.StandardCharsets;
+import java.util.ArrayList;
+import java.util.Collections;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+import java.util.SortedSet;
+
+import org.junit.Assert;
+import org.junit.Test;
+
+import com.puppycrawl.tools.checkstyle.AbstractModuleTestSupport;
+import com.puppycrawl.tools.checkstyle.DefaultConfiguration;
+import com.puppycrawl.tools.checkstyle.api.CheckstyleException;
+import com.puppycrawl.tools.checkstyle.api.FileText;
+import com.puppycrawl.tools.checkstyle.api.LocalizedMessage;
+import com.puppycrawl.tools.checkstyle.utils.CommonUtil;
+
+public class OrderedPropertiesCheckTest extends AbstractModuleTestSupport {
+
+    @Override
+    protected String getPackageLocation() {
+        return "com/puppycrawl/tools/checkstyle/checks/orderedproperties";
+    }
+
+    /**
+     * Tests the ordinal work of a check.
+     * The sorted file should look like this
+     * A=2=3
+     * key1=value1
+     * # comment
+     * key2=value2
+     * key3=value3
+     * key4=
+     */
+    @Test
+    public void testDefault() throws Exception {
+        final DefaultConfiguration checkConfig = createModuleConfig(OrderedPropertiesCheck.class);
+        final String[] expected = {
+            "3: " + getCheckMessage(MSG_KEY, "key1", "key2"),
+            "6: " + getCheckMessage(MSG_KEY, "A", "key4"),
+        };
+        verify(checkConfig, getPath("InputOrderedProperties.properties"), expected);
+    }
+
+    @Test
+    public void testNotFoundKey() throws Exception {
```

Comment:
Question wasn't responded to at https://github.com/checkstyle/checkstyle/pull/6343#discussion_r245652053
> Why can't this use an input file instead of hacking the check?

Reply:
OK, I will rework this.

Reply:
Test deleted. Done.